src/scripts/main.js

```diff
@@ -1,3 +1,105 @@
 'use strict';
 
 // write code here
+const container = document.querySelector('.container');
+const appendRowButton = container.querySelector('.append-row');
+const removeRowButton = container.querySelector('.remove-row');
+const appendColumnButton = container.querySelector('.append-column');
+const removeColumnButton = container.querySelector('.remove-column');
+
+const appendRow = (tableRows) => {
+  const newRow = document.createElement('tr');
+
+  for (
+    let i = 0;
+    i < tableRows.lastElementChild.querySelectorAll('td').length;
+    i++
+  ) {
+    const newCell = document.createElement('td');
+
+    newRow.appendChild(newCell);
+  }
+  tableRows.appendChild(newRow);
+};
+
+const removeRow = (tableRows) => {
+  tableRows.lastElementChild.remove();
+};
+
+const appendColumn = (tableRows) => {
+  tableRows.querySelectorAll('tr').forEach((elem) => {
+    const newTableData = document.createElement('td');
+
+    elem.appendChild(newTableData);
+  });
+};
+
+const removeColumn = (tableRows) => {
+  tableRows.querySelectorAll('tr').forEach((elem) => {
+    elem.lastElementChild.remove();
+  });
+};
+
+container.addEventListener('click', (e) => {
+  const clickedButton = e.target.closest('button');
+
+  if (!clickedButton) {
+    return;
+  }
+
+  const MAX_ROWS = 10;
+  const MAX_COLUMNS = 10;
+  const MIN_ROWS = 2;
+  const MIN_COLUMNS = 2;
+
+  const tableRows = container.querySelector('tbody');
+  let rowsCount = tableRows.children.length;
+  let columnsCount = tableRows.firstChild.children.length;
```

Comment:
The `rowsCount` and `columnsCount` are recalculated every time a button is clicked. You could improve performance by calculating these values only when they are needed, i.e., when a row or column is added or removed.